src/common/components/paymentMethods/creditCardForm/creditCardForm.component.js

```diff
@@ -131,24 +119,42 @@ class CreditCardController {
   savePayment(){
     this.creditCardPaymentForm.$setSubmitted();
     if(this.creditCardPaymentForm.$valid){
-      let ccpCreditCardNumber = this.paymentMethod && !this.creditCardPayment.cardNumber ? this.paymentMethod['card-number'] : new (this.paymentValidationService.ccp.CardNumber)(this.creditCardPayment.cardNumber).encrypt();
-      let ccpSecurityCode = this.paymentMethod ? null : new (this.paymentValidationService.ccp.CardSecurityCode)(this.creditCardPayment.securityCode).encrypt();
-      this.onPaymentFormStateChange({
-        $event: {
-          state: 'loading',
-          payload: {
-            creditCard: {
-              address: this.useMailingAddress ? undefined : this.creditCardPayment.address,
-              'card-number': ccpCreditCardNumber,
-              'cardholder-name': this.creditCardPayment.cardholderName,
-              'expiry-month': this.creditCardPayment.expiryMonth,
-              'expiry-year': this.creditCardPayment.expiryYear,
-              ccv: ccpSecurityCode
-            },
-            paymentMethodNumber: this.creditCardPayment.cardNumber ? this.creditCardPayment.cardNumber.slice(-4) : false
-          }
-        }
-      });
+      this.tsysService.getManifest()
+        .mergeMap(data => {
+          cruPayments.creditCard.init(this.envService.get(), data.deviceId || '88812128320102', data.manifest); //TODO: remove hard coded deviceId when added to the manifest api endpoint
+          return this.paymentMethod && !this.creditCardPayment.cardNumber ?
+            Observable.of({ tsepToken: this.paymentMethod['card-number'], maskedCardNumber: this.paymentMethod['card-number'] }) : // Send masked card number when card number is not updated
+            cruPayments.creditCard.encrypt(this.creditCardPayment.cardNumber, this.creditCardPayment.securityCode, this.creditCardPayment.expiryMonth, this.creditCardPayment.expiryYear);
```

Comment:
Will `this.creditCardPayment.securityCode` be null when it needs to be? The cases where we won't have one would be:
* Creating a new credit card via self-service
* Editing an existing credit card to a new card number via self-service
* Giving recurring donations with a credit card during checkout
* Giving a mix of recurring and one-time donations with a credit card during checkout